common/thorhelper/thorsoapcall.cpp

```diff
@@ -941,46 +938,102 @@ class CWSCHelper : implements IWSCHelper, public CInterface
+
+            StringBuffer value;
+            getSecret(value, secretName, "url", false);
+            if (!value.length())
+                throw MakeStringException(0, "%sCALL HTTP-CONNECT SECRET (%s) url required", wscType == STsoap ? "SOAP" : "HTTP", secretName.str());
+
+            UrlListParser urlListParser(value);
+            getSecret(value.clear(), secretName, "username", false);
+            if (strchr(value, ':'))
+                throw MakeStringException(0, "%sCALL HTTP-CONNECT SECRET (%s) username contains illegal colon", wscType == STsoap ? "SOAP" : "HTTP", secretName.str());
+            if (value.length())
             {
-                numRowThreads = 1;
+                value.append(':');
+                getSecret(value, secretName, "password", false);
+            }
+            urlListParser.getUrls(urlArray, value.str());
 
-                numUrlThreads = helper->numParallelThreads();
-                if (numUrlThreads == 0)
-                    numUrlThreads = 1;
-                else if (numUrlThreads > MAXWSCTHREADS)
-                    numUrlThreads = MAXWSCTHREADS;
+            getSecret(proxyAddress, secretName, "proxy", false);
 
-                numRecordsPerBatch = 1;
-            }
-            else
+            if (secretExists(secretName, "tls.crt"))
             {
-                unsigned totThreads = helper->numParallelThreads();
-                if (totThreads < 1)
-                    totThreads = 2; // default to 2 threads
-                else if (totThreads > MAXWSCTHREADS)
-                    totThreads = MAXWSCTHREADS;
-
-                numUrlThreads = (numUrls < totThreads)? numUrls: totThreads;
-
-                numRowThreads = totThreads / numUrlThreads;
-                if (numRowThreads < 1)
-                    numRowThreads = 1;
-                else if (numRowThreads > MAXWSCTHREADS)
-                    numRowThreads = MAXWSCTHREADS;
-
-                numRecordsPerBatch = helper->numRecordsPerBatch();
-                if (numRecordsPerBatch < 1)
-                    numRecordsPerBatch = 1;
+                getSecret(value.clear(), secretName, "tls.key", false);
+                if (value.isEmpty())
+                    throw MakeStringException(0, "%sCALL HTTP-CONNECT SECRET (%s) provides client certificate but not private key", wscType == STsoap ? "SOAP" : "HTTP", secretName.str());
+
+                clientCert.setown(new ClientCertificate());
+                clientCert->privateKey.set(value);
+                getSecret(value.clear(), secretName, "tls.crt");
+                clientCert->certificate.set(value);
+                customClientCert = true;
             }
         }
         else
+        {
+            UrlListParser urlListParser(hosts);
+            urlListParser.getUrls(urlArray);
+            proxyAddress.set(s.setown(helper->getProxyAddress()));
+        }
+
+        if ((numUrls = urlArray.ordinality()) == 0)
```

Comment:
trivial:  The code existed already, but (arguably) clearer to assign to numUrls first.